src/Http.cpp

```diff
@@ -237,10 +240,143 @@ auto to_string(
     }
 }
 
-auto to_enum(
-    uint32_t code) -> StatusCode
+auto to_enum(int32_t code) -> StatusCode
 {
-    return static_cast<StatusCode>(code);
+    switch (code)
+    {
+        case 100:
+            return StatusCode::HTTP_100_CONTINUE;
+        case 101:
+            return StatusCode::HTTP_101_SWITCHING_PROTOCOLS;
+        case 102:
+            return StatusCode::HTTP_102_PROCESSING;
+        case 103:
+            return StatusCode::HTTP_103_EARLY_HINTS;
+
+        case 200:
+            return StatusCode::HTTP_200_OK;
+        case 201:
+            return StatusCode::HTTP_201_CREATED;
+        case 202:
+            return StatusCode::HTTP_202_ACCEPTED;
+        case 203:
+            return StatusCode::HTTP_203_NON_AUTHORITATIVE_INFORMATION;
+        case 204:
+            return StatusCode::HTTP_204_NO_CONTENT;
+        case 205:
+            return StatusCode::HTTP_205_RESET_CONTENT;
+        case 206:
+            return StatusCode::HTTP_206_PARTIAL_CONTENT;
+        case 207:
+            return StatusCode::HTTP_207_MULTI_STATUS;
+        case 208:
+            return StatusCode::HTTP_208_ALREADY_REPORTED;
+        case 226:
+            return StatusCode::HTTP_226_IM_USED;
+
+        case 300:
+            return StatusCode::HTTP_300_MULTIPLE_CHOICES;
+        case 301:
+            return StatusCode::HTTP_301_MOVED_PERMANENTLY;
+        case 302:
+            return StatusCode::HTTP_302_FOUND;
+        case 303:
+            return StatusCode::HTTP_303_SEE_OTHER;
+        case 304:
+            return StatusCode::HTTP_304_NOT_MODIFIED;
+        case 305:
+            return StatusCode::HTTP_305_USE_PROXY;
+        case 306:
+            return StatusCode::HTTP_306_SWITCH_PROXY;
+        case 307:
+            return StatusCode::HTTP_307_TEMPORARY_REDIRECT;
+        case 308:
+            return StatusCode::HTTP_308_PERMANENT_REDIRECT;
+
+        case 400:
```

Comment:
Is it worth mapping unknown 4xx and 5xx codes to 400 or 500 (respectively)?

Reply:
Good question! I've double checked and updated against the current codes the IANA have, so I think we're okay to let anything not handled be unknown.